lib/http/preprocessors.js

```diff
@@ -67,7 +68,7 @@ const authHandler = ({ Sessions, Users, Auth, bcrypt }, context) => {
     return authBySessionToken(authHeader.substring(7), () => reject(Problem.user.authenticationFailed()));
 
   // Basic Auth, which is allowed over HTTPS only:
-  } else if (isPresent(authHeader) && authHeader.startsWith('Basic ')) {
+  } else if (!oidc.isEnabled() && isPresent(authHeader) && authHeader.startsWith('Basic ')) {
```

Comment:
Should we throw a new `Problem.user.ssoEnabled` that should tell user that we don't support Basic authentication when SSO is enabled? or mentioning this in user docs is sufficient

cc: @lognaturel @matthew-white

Reply:
A new Problem sounds nice if it's easy to do. As is, if someone tries to use Basic Auth on a server that uses SSO, they won't receive a 401. Instead, they'll be allowed to access endpoints that don't require auth (e.g., /v1/projects), or they'll receive a 403. In contrast, if they try to use Basic Auth without using HTTPS, they'll immediate receive a 401 with a specific explanation. I think we could do something similar for SSO. I think that change could also come in a follow-up PR.